### PR TITLE
CONSOLE-3611: User can configure the install of Operators that authenticate using STS

### PR DESCRIPTION
**Pull request by gallettilance**
Only display warning in the subscription page when cluster is STS and the operator supports it.

Subscription install mode default to Manual when cluster is STS and the operator supports it.

There is a warning for manual subscriptions.

Only display role ARN field if cluster is STS and operator supports it.

Install button is grayed out if the role ARN is empty, the cluster is in STS mode, and the operator supports STS.

Add role ARN env var to the subscription config set by the user during install.

[Screencast from 2023-07-03 10-28-19.webm](https://github.com/openshift/console/assets/20077170/ef99e23a-bb01-49df-8eaf-a24af1591ccc)

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 1]
```diff
@@ -71,6 +72,7 @@ import {
 import { installedFor, supports, providedAPIsForOperatorGroup, isGlobal } from '../operator-group';
 
 export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> = (props) => {
+  const [filterText, setFilterText] = React.useState('');
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:75]: Nit:  names are too generic.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 2]
```diff
@@ -719,6 +730,24 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
         <div className="row">
           <div className="col-xs-6">
             <>
+              {(
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:733]: Needs a condition so it only renders when needed.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 3]
```diff
@@ -389,7 +391,14 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
         startingCSV: channels.find((ch) => ch.name === selectedUpdateChannel).currentCSV,
         channel: selectedUpdateChannel,
         installPlanApproval: selectedApproval,
-      },
+        config: {
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:394]: Make conditional.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 4]
```diff
@@ -719,6 +730,24 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
         <div className="row">
           <div className="col-xs-6">
             <>
+              {(
+                <div className="form-group">
+                <fieldset>
+                  <div className="co-toolbar__item">
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:736]: Add `<Label>`

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 5]
```diff
@@ -252,11 +260,19 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
     selectedTargetNamespace,
   );
 
+  const tokenType = 'role ARN';
+  const tokenizedAuth = getQueryArgument('tokenizedAuth');
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:264]: Can add `tokenizedAuth` to line 89 (and update lines 93 and 99 to use it as well, which several of us missed when @sg00dwin made those changes).

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 6]
```diff
@@ -252,11 +260,19 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
...
+  const [showWarn, setShowWarn] = React.useState(true);
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:265]: Would be great if this hook could live closer to the beginning of the component per best practices.  And the name is a bit generic.  Maybe `showSTSWarn`?

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 7]
```diff
@@ -735,9 +763,42 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
...
+              {t(
+                'olm~This cluster is using AWS Security Token Service to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, you will need to provide a role ARN (with an attached policy) during installation. Manual subscriptions are highly recommended as steps should be taken prior to upgrade to ensure that the permissions required by the next version are properly accounted for in the role. Please see the operator description for more details.',
+              )}
+            </p>
+          </Alert>
+        )}
         <div className="row">
           <div className="col-xs-6">
             <>
+              {
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:784]: Needs conditional.

**Review by rhamilto — COMMENTED** [review]
Not sure if you're ready for another review, but I gave it a quick look anyhow.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx (hunk 5 above)
Comment (gallettilance) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:264]: sorry not sure I understand what the updates to 93 and 99 would look like

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 8]
```diff
@@ -91,6 +99,10 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
   const version = getQueryArgument('version');
   const [updateVersion, setUpdateVersion] = React.useState(version || currentLatestVersion);
 
+  const tokenType = 'role ARN';
+  const tokenizedAuth = getQueryArgument('tokenizedAuth');
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:103]: Sorry.  I should have been more clear.  

Remove this line and lines 93 and 99 and line 89 becomes 
`const { catalogNamespace, channel, pkg, tokenizedAuth, version } = getURLSearchParams();`. This simply deduplicates the multiple `getQueryArgument` calls and consolidates them in to the existing `getURLSearchParams` one (which returns an object of parameters and not a single parameter).

**Comment by rhamilto** [comment]
@gallettilance, please investigate the Cypress integration test failures.  It's possible your changes in this PR broke several tests.

**Comment by jhadvig** [comment]
@gallettilance we need to rebase since https://github.com/openshift/console/pull/12778 got merged

**Comment by jcaianirh** [comment]
Assigning other approvers:

QE Approver: 
/assign @yapei
Docs Approver:
 /assign @opayne1
PX Approver:
/assign @RickJWagner

**Review by rhamilto — APPROVED** [review]
/lgtm

**Review by sg00dwin — APPROVED** [review]

**Comment by jhadvig** [comment]
QE Approver: 
/assign @yanpzhan 
Docs Approver:
 /assign @ahardin-rh

**Comment by yanpzhan** [comment]
@gallettilance could you rebase this pr so that it contains code from pr12980?

**Comment by gallettilance** [comment]
@yanpzhan  done

**Pull request merged**